stage1/app-rm/app-rm.go

```diff
@@ -86,24 +113,54 @@ func main() {
 
 	for _, p := range appServicePaths {
 		if err := os.Remove(p); err != nil && !os.IsNotExist(err) {
-			log.FatalE("error removing app service file", err)
+			return fmt.Errorf("error removing app service file: %s", err)
 		}
 	}
 
-	args = enterCmd
-	args = append(args, "/usr/bin/systemctl")
-	args = append(args, "daemon-reload")
+	// TODO(sur): find all RW cgroups exposed for this app and clean them up
 
-	cmd = exec.Cmd{
-		Path: args[0],
-		Args: args,
+	// last cleaning steps are performed after entering pod context
+	tasks := [][]string{
+		// inception: perform phase1 before proceeding
+		{"/app-rm", "--phase=1", fmt.Sprintf("--app=%s", appName), fmt.Sprintf("--debug=%t", debug)},
+		// all cleaned-up, let systemd reload and forget about this app
+		{"/usr/bin/systemctl", "daemon-reload"},
 	}
+	for _, cmdLine := range tasks {
+		args := append(enterCmd, cmdLine...)
+		cmd = exec.Cmd{
+			Path: args[0],
+			Args: args,
+		}
 
-	if out, err := cmd.CombinedOutput(); err != nil {
-		log.Fatalf("%q failed at daemon-reload:\n%s", appName, out)
+		if out, err := cmd.CombinedOutput(); err != nil {
+			return fmt.Errorf("%q removal failed:\n%s", appName, out)
+		}
 	}
+	return nil
+}
 
-	// TODO unmount all the volumes
+// cleanupPhase1 inspects pod systemd-pid1 mountinfo to find all remaining
```

Comment:
`// cleanupPhase1 is executed in stage1 and inspects the pod's systemd-pid1 moutinfo ....`